model-optimizer/mo/front/tf/graph_utils.py

```diff
@@ -128,6 +126,11 @@ def add_convolution_to_swap_xy_coordinates(graph: Graph, input_node: Node, coord
 
     input_reshape_4d_node = create_op_node_with_second_input(graph, Reshape, int64_array([-1, 1, 1, coordinates_size]),
                                                              dict(name=input_node.name + '/reshape_4d'), input_node)
+
+    # we need to import these functions here to avoid circular dependent imports
```

Comment:
I don't understand how we introduce circular dependency and why it is solved this way. Please, elaborate on that.

Reply:
We have circular dependencies `extensins\middle\InsertLayoutPropagationTransposes.py` and `mo\front\tf\graph_utils.py`. In `add_convolution_to_swap_xy_coordinates(...)` function from second file we use `mark_input_as_in_correct_layout(...)` and `mark_output_as_in_correct_layout(...)` functions from fist file. And in first file we use `create_op_node_with_second_input(...)` function from second file. Now more direct solution implemented.